lib/AST/Module.cpp

```diff
@@ -1069,14 +1163,46 @@ SourceFile::getImportedModules(SmallVectorImpl<ModuleDecl::ImportedModule> &modu
     else
       requiredKind = ModuleDecl::ImportFilterKind::Private;
 
-    if (filter.contains(requiredKind))
+    if (filter.contains(requiredKind)) {
+      // Filter out duplicate imports; they have no semantic effect.
+      auto iter = std::find_if(modules.begin(), modules.end(),
```

Comment:
Nitpick: `llvm::find_if` (and `llvm::copy` elsewhere).

Reply:
Sure. What’s the difference?

Reply:
These two are just easier to read (since you don't need to pass the iterators separately), but some of the other STL covers do extra checking (e.g. to make sure you aren't relying on iteration order for unordered containers).